### DISCUSSION TITLE
Provide Winget as a package manager option

### DISCUSSION THREAD

**Discussion by umitseyhan75** [samuel-lucas6/Kryptor#52] [Ideas]
People getting more and more used to it, so it would be good idea.

**Comment by samuel-lucas6** [comment]
That would be nice, but I have no knowledge of winget and would prefer if someone else would maintain the package if it requires updating every release.

Also, as Kryptor is already on Chocolatey and Scoop for Windows, a higher priority is getting it on some macOS and Linux package managers [like age](https://github.com/FiloSottile/age#installation). Again, I don't have the relevant experience, don't use these package managers, and keeping several packages up to date makes a release even more time consuming.

**Reply by umitseyhan75** [reply]
I do not have the relevant knowledge as well but as a Winget user I do know one thing that is Winget downloads packages/installers from where ever devs upload. From Github releases to Sourceforge, from Fosshub to your own hosting machine. So I believe it would not be a hassle to maintain it.

[Here](https://learn.microsoft.com/en-us/windows/package-manager/package/l) is a link if you would like to check it out.

**Reply by samuel-lucas6** [reply]
Thanks, I'll take a look at some point, but I won't make any promises I'm afraid.

**Comment by umitseyhan75** [comment]
Regarding this discussion, I asked Microsoft Copilot about it and here is the answer it gave me:

> To publish software on Winget (Windows Package Manager), a software engineer needs to follow these steps:
> 
> 1. Create a Package Manifest: This is a YAML file that describes the application. The manifest includes details like the software name, version, installer type, and other metadata1.
> 
> 2. Validate the Manifest: Use the winget validate command to ensure the manifest is correctly formatted.
> 
> 3. Test the Manifest: Run the manifest through the Windows Sandbox to test the installation process.
> 
> 4. Submit the Manifest: Clone the Winget repository on GitHub, add your manifest, and submit a pull request. The Winget team will review and validate your submission2.
> 
> 5. Monitor and Maintain: Once accepted, monitor the software for any issues and update the manifest as needed.

Sounds not so hard. Here is a documentation from Microsoft: https://learn.microsoft.com/en-us/windows/package-manager/package/manifest?tabs=minschema%2Cversion-example

**Reply by samuel-lucas6** [reply]
The trouble is if it requires a new PR every update. There are also two supported package managers for Windows already, whereas there are none for macOS and very few for Linux. I'd rather prioritise something like Homebrew, but every package I maintain just makes getting a new release out harder.

**Reply by umitseyhan75** [reply]
Can't you set up an automation for it?

**Reply by samuel-lucas6** [reply]
Yes, it sounds like it can be automated after the initial submission.

**Reply by umitseyhan75** [reply]
Sounds great! New release and Winget will be big enough news.

**Reply by samuel-lucas6** [reply]
I was going to get a new release out today, but there's unfortunately an issue with the libsodium NuGet package that affects the macOS binaries, so I'll have to wait until that's fixed.
